snuba/api.py

```diff
@@ -182,14 +183,23 @@ def parse_request_body(request):
         raise BadRequest(str(error)) from error
 
 
-def validate_request_content(body, schema, timer):
+def validate_request_content(body, extensions_schema, timer):
+    schema = schemas.get_composite_schema([
+        GENERIC_QUERY_SCHEMA,
+        extensions_schema,
+    ])
+
     try:
         schemas.validate(body, schema)
     except jsonschema.ValidationError as error:
         raise BadRequest(str(error)) from error
 
+    query = {key: body.pop(key) for key in GENERIC_QUERY_SCHEMA['properties'].keys() if key in body}
```

Comment:
What's the semantics you have in mind for queries that have dataset specific schemas? Should each one have a disjoint set of properties ?

Reply:
> Should each one have a disjoint set of properties ?

For the short term, they'll have to be disjoint to preserve backwards compatibility. I don't like that very much. I think I already answered your question in the PR description, though?

Reply:
Well, the description did not talk about how the dataset extension are merged together. I like starting with a disjoint set of properties in the short term.

Reply:
Maybe I misunderstood the question then. I thought you were asking about longer-term plan (in addition to the short term one) where I mentioned how I'd like it to work (so that it's not limited to disjoint namings moving forward):

> ```python
> {
>   "query": …,
>   "extensions": …,  # may contain another level for each "extension", e.g. {"timeseries": …, }
> }
> ```